src/Persisters/Flysystem.php

```diff
@@ -182,12 +152,9 @@ public function load() : Persistable
      */
     public function __toString() : string
     {
-        $filesystem = 'Flysystem';
-
-        if (method_exists($this->filesystem, 'getAdapter')) {
-            $filesystem .= ' <' . Params::toString($this->filesystem->getAdapter()) . '>';
-        }
-
-        return "{$filesystem} (path: '{$this->path}', serializer: {$this->serializer})";
+        return "Flysystem (path: {$this->path}, filesystem: "
+            . Params::toString($this->filesystem) . ', history: '
+            . Params::toString($this->history) . ', serializer: '
+            . "{$this->serializer})";
```

Comment:
I think it's more informative to have the _adapter_ (`League\Flysystem\AdapterInterface`) used not the `FilesystemOperator` wrapper class. As `FilesystemOperator` (and its predecessor `FilesystemInterface`) doesn't enforce access to it, I had the runtime reflection to check.

My aim was to have the string output be something along the lines of:

```
Flysystem (path: path/to/test.persistable, filesystem: InMemory, history: false, serializer: Native)
```

rather than: 
```
Flysystem (path: path/to/test.persistable, filesystem: Filesystem, history: false, serializer: Native)
```

Reply:
Yeah I liked the way you had it before, will probably put back

Reply:
I admit it looked a little messy, I did abstract this logic into a helper method in a branch i have locally: 

```php
Params::property(object $subject, string $property, string $fallback = ''): mixed;
```
so could be used as:

```php
Params::property($this->filesystem, 'adapter')
```

which internally uses reflection if the property is `protected` or `private`, the fallback value is used if the property doesn't exist on the subject.

Can push it up to a branch later and you can decide if you like it or not 👍